src/hive/rework/core.cljs

```diff
@@ -128,18 +128,24 @@
    - a vector whose first element is a function and the rest are its argument.
      The function will be executed and its return value is passed to transact!
      again. Useful for keeping functions side-effect free
-   - a delay-effect whose content will be executed and passed to transact!
-     again. Useful for keeping function side-effect free when doing Js interop
+   - a delay whose content will be forced and passed to transact! again.
+     Useful for keeping function side-effect free when doing Js interop
+   - a Js Promise, whose return value will be transact. Useful for doing
+     asynchronous Js calls without having to convert it to a core async channel
```

Comment:
I have to yet learn the concept of a _channel_ in core async.

Reply:
@mehdisadeghi I just think of them as very powerful queues. They are of course more than that but it helps to think of them in term of asynchronous sequences ;)